lib/validations.js

```diff
@@ -340,7 +340,20 @@ function validateCustom(attr, conf, err, options, done) {
     done = options;
     options = {};
   }
-  conf.customValidator.call(this, err, done);
+
+  if (conf.customValidator.length <= 1) {
```

Comment:
Our usual convention in other places where we support both callback/promise based functions, is to check for the return type of the function, instead of relying on the number of arguments.

```js
var result = conf.customValidator(...);
if (result.then && typeof result.then === 'function') {
  // ...
}
```

My concern is that a custom validator accepting a single argument is a valid use case, see https://github.com/strongloop/loopback-datasource-juggler/pull/1229/files#diff-a1a277b8e694ce811867b2ac3e064ff8R626

```js
       User.validate('email', function(err) {
         if (this.email === 'hello') err();
       }, {code: 'invalid-email'});
```